### PR TITLE
Do not try to parse huge compressed archive files

### PR DESCRIPTION
**Pull request by hughsie**
We don't want to allow DoS-ing the daemon by trying to decompress a huge sparse
cabinet archive.

### CODE REVIEW

File: src/fu-common-cab.c [hunk 1]
```diff
@@ -227,12 +227,49 @@ fu_common_store_from_cab_folder (AsStore *store, GCabFolder *cabfolder, GError *
 	return TRUE;
 }
 
+typedef struct {
+	guint64		 size_total;
+	guint64		 size_max;
+	GError		*error;
+} FooHelper;
```
Comment (superm1) [inline src/fu-common-cab.c]: Not really a fan of this name.  How about something like CabCBHelper or something more descriptive?

File: src/fu-common-cab.c [hunk 2]
```diff
@@ -227,12 +227,49 @@ fu_common_store_from_cab_folder (AsStore *store, GCabFolder *cabfolder, GError *
...
+
 static gboolean
 as_cab_store_file_cb (GCabFile *file, gpointer user_data)
 {
+	FooHelper *helper = (FooHelper *) user_data;
```
Comment (superm1) [inline src/fu-common-cab.c]: same

File: src/fu-engine.h [hunk 3]
```diff
@@ -35,7 +35,7 @@ G_BEGIN_DECLS
 #define FU_TYPE_ENGINE (fu_engine_get_type ())
 G_DECLARE_FINAL_TYPE (FuEngine, fu_engine, FU, ENGINE, GObject)
 
-#define		FU_ENGINE_FIRMWARE_SIZE_MAX		(32 * 1024 * 1024) /* bytes */
+#define		FU_ENGINE_FIRMWARE_SIZE_MAX		(512 * 0x100000) /* bytes * Mb */
```
Comment (superm1) [inline src/fu-engine.h]: Why keep around both `FU_ENGINE_FIRMWARE_SIZE_MAX `and `FU_COMMON_ARCHIVE_SIZE_MAX`?  It seems like you should be able to get away with 1 or the other.  I know they're set to different values right now, but so why?
If both should stick around a comment around their intent might be useful for the future.

File: src/fu-common-cab.c (hunk 1 above)
Comment (hughsie) [reply src/fu-common-cab.c]: Gahh! I was supposed to change that, good catch. Fixed.

File: src/fu-engine.h (hunk 3 above)
Comment (hughsie) [reply src/fu-engine.h]: I pondered about moving the define to fu-common-cab.h, but you're right, we can just use the old define name. Fixed.

**Comment by hughsie** [comment]
Does 1d45125 work for you?

**Comment by superm1** [comment]
Yep looks great now!

**Pull request merged**